hibernate-core/src/main/java/org/hibernate/query/sql/internal/NativeSelectQueryPlanImpl.java

```diff
@@ -140,8 +144,12 @@ public NativeSelectQueryPlanImpl(
 							type = StandardBasicTypes.OBJECT_TYPE;
 						}
 
-						final JdbcMapping jdbcMapping = ( (BasicValuedMapping) type ).getJdbcMapping();
-						final JdbcParameterImpl jdbcParameter = new JdbcParameterImpl( jdbcMapping );
+						BasicValuedMapping basicValuedMapping = (BasicValuedMapping) type;
+						final JdbcMapping jdbcMapping = basicValuedMapping.getJdbcMapping();
+						final JdbcParameterImpl jdbcParameter = new JdbcParameterImpl(
```

Comment:
Really, the mapping here is only need to get the converter (if one).  I wonder if we should instead simply pass in the converter?

Reply:
I moved the `MappingModelExpressable` to the `QueryParameterBinding` now like we discussed to handle the conversion during binding of the parameter value